tensorflow/core/grappler/optimizers/BUILD

```diff
@@ -1158,7 +1158,10 @@ tf_cuda_cc_test(
 cc_library(
     name = "generic_layout_optimizer",
     srcs = ["generic_layout_optimizer.cc"],
-    hdrs = ["generic_layout_optimizer.h"],
+    hdrs = ["generic_layout_optimizer.h"]    
```

Comment:
I am not sure that this is best way to fix this.

The existing //tensorflow/core/platform:stream_executor dependency was sufficient prior to this week's sync so that seems to indicate that the root cause is elsewhere.

A quick check revealed that the dependencies for //tensorflow/core/platform:stream_executor was changed upstream   so that is probably a better place to fix this.

I would suggest using bazel cquery to list the deps for that target in this branch vs. the current tip of develop-upstream and compare.  That might reveal a better fix.

Fixing it there might eliminate the need for the change in tensorflow/core/kernels/BUILD.

Reply:
@rsanthanam-amd  See if you like these changes

Reply:
much better, thanks.